spec/services/order_invoice_comparator_spec.rb

```diff
@@ -115,19 +115,241 @@
+          it "first name" do
+            ship_address.update!(firstname: "Jane")
+            order.reload
+            expect(subject).to be true
+          end
+
+          it "last name" do
+            ship_address.update!(lastname: "Jones")
+            order.reload
+            expect(subject).to be true
+          end
+
+          it "address (1)" do
+            ship_address.update!(address1: "Rue du Fromage 66")
+            order.reload
+            expect(subject).to be true
+          end
+
+          it "address (2)" do
+            ship_address.update!(address2: "South by Southwest")
+            order.reload
+            expect(subject).to be true
+          end
+
+          it "city" do
+            ship_address.update!(city: "Antibes")
+            order.reload
+            expect(subject).to be true
+          end
+
+          it "zipcode" do
+            ship_address.update!(zipcode: "04229")
+            order.reload
+            expect(subject).to be true
+          end
+
+          it "phone" do
+            ship_address.update!(phone: "111-222-333")
+            order.reload
+            expect(subject).to be true
+          end
+
+          it "company" do
+            ship_address.update!(company: "A Company Name")
+            order.reload
+            expect(subject).to be true
+          end
+        end
+
+        context "customer changes on" do
+          let(:customer){ order.customer }
+
+          context "code" do
+            it "return false" do
+              customer.update!(code: 98_754)
+              order.reload
+              expect(subject).to be false
+            end
+          end
+
+          context "email" do
+            it "return false" do
+              customer.update!(email: "customer@email.org")
+              order.reload
+              expect(subject).to be false
+            end
+          end
+        end
+
+        context "payment changes on" do
+          let(:payment) { create(:payment, order_id: order.id) }
+          context "amount" do
+            it "returns true" do
+              payment.update!(amount: 222)
+              order.reload
+              expect(subject).to be true
+            end
+          end
+
+          context "payment method" do
+            let(:payment_method) { create(:payment_method) }
+
+            it "returns true" do
+              payment.update!(payment_method_id: payment_method.id)
+              order.reload
+              expect(subject).to be true
+            end
+          end
+
+          context "payment changes on" do
+            context "state" do
+              it "returns true" do
+                expect {
+                  payment.complete
+                }.to change { payment.state }.from("checkout").to("completed")
+
+                order.reload
+                expect(subject).to be true
+              end
+            end
+          end
+        end
```

Comment:
... but it also passes here, which is the `generate new invoice section`. Any idea why this is happening @abdellani ?

Reply:
Yes, this is not supposed to happen.
I'll inspect the issue.

Reply:
This is happening because `order.payment_total` changes from 0 to `0.4575e2` after running  `payment.complete`.

Reply:
Discussed and fixed on our live chat, resolving :pray: